File: csharp/prime-factors/PrimeFactors.cs
using System;
using System.Collections.Generic;

public static class PrimeFactors {
    public static long[] Factors (long number) {
        List<long> list = new List<long> ();
        long currentPrime = 2;
        while (number != 1) {
            if (number % currentPrime == 0) {
                list.Add (currentPrime);
                number = number / currentPrime;
            } else {
                do {
                    currentPrime++;
                } while (!IsPrime (currentPrime));
            }
        }
        return list.ToArray ();
    }

    public static bool IsPrime (long val) {
        bool f = true;
        for (int i = 2; i < Math.Sqrt (val) && f; i++) {
            if (val % i == 0) {
                f = false;
            }
        }
        return f;
    }
}

File: cpp/prime-factors/prime_factors.cpp
#include "prime_factors.h"
#include <cmath>

namespace prime_factors
{

    bool is_prime(long long n)
    {
        for (long long i = 2; i <= (long long)sqrt(n); i++)
        {
            if (n % i == 0)
            {
                return false;
            }
        }

        return true;
    }

    std::vector<long long> of(long long n)
    {
        std::vector<long long> result;
        long long current = 2;

        while (n != 1)
        {
            if (n % current == 0)
            {
                result.push_back(current);
                n = n / current;
            }
            else
            {
                do
                {
                    current++;
                } while (!is_prime(current));
            }
        }

        return result;
    }

} // namespace prime_factors
